src/SwedbankPay.Sdk/Amount.cs

```diff
@@ -1,50 +1,113 @@
 ﻿using System;
-
-using Newtonsoft.Json;
-
+using System.Globalization;
+
 namespace SwedbankPay.Sdk
 {
-    public class Amount
-    {
-        [JsonConstructor]
-        private Amount(decimal value)
-        {
-            if (value < 0)
-                throw new ArgumentOutOfRangeException(nameof(value), "Negative amount value is not allowed.");
-            Value = decimal.ToInt32(value);
-        }
-
-
-        internal Amount(long value)
-        {
-            Value = value;
-        }
-
-
-        public long Value { get; }
-
-
-        public static Amount FromDecimal(decimal amount)
-        {
-            return new Amount(amount * 100);
-        }
-
-
-        public static Amount FromInt(long amount)
-        {
-            return new Amount(amount * 100);
-        }
-
-
-        public static decimal ToDecimal(Amount amount)
-        {
-            return (decimal)amount.Value / 100;
-        }
-
-
-        public override string ToString()
-        {
-            return Value.ToString();
+    public class Amount : IEquatable<Amount>, IComparable<Amount>, IComparable

```

Comment:
Codacy found an issue: [When implementing IComparable<T> or IComparable, you should also override <, >, <=, >=, ==, !=.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)